zulip/zulip/send.py

```diff
@@ -71,28 +71,32 @@ def main() -> int:
     if len(options.recipients) == 0 and not (options.stream and options.subject):
         parser.error('You must specify a stream/subject or at least one recipient.')
 
-    client = zulip.init_from_options(options)
-
-    if not options.message:
-        options.message = sys.stdin.read()
-
-    if options.stream:
-        message_data = {
-            'type': 'stream',
-            'content': options.message,
-            'subject': options.subject,
-            'to': options.stream,
-        }
-    else:
-        message_data = {
-            'type': 'private',
-            'content': options.message,
-            'to': options.recipients,
-        }
-
-    if not do_send_message(client, message_data):
-        return 1
-    return 0
+    try:
+        client = zulip.init_from_options(options)
+
+        if not options.message:
+            options.message = sys.stdin.read()
+
+        if options.stream:
+            message_data = {
+                'type': 'stream',
+                'content': options.message,
+                'subject': options.subject,
+                'to': options.stream,
+            }
+        else:
+            message_data = {
+                'type': 'private',
+                'content': options.message,
+                'to': options.recipients,
+            }
+
+        if not do_send_message(client, message_data):
+            return 1
+        return 0
+    except zulip.InvalidCredentialsError as error:
+        logging.error(error)
+        return 0
```

Comment:
We should return a non-zero value here. 0 implies the process went smoothly, without errors.